restructuredtext_lint/sphinx.py

```diff
@@ -0,0 +1,63 @@
+from __future__ import absolute_import
+
+try:
+    import sphinx  # noqa
+
+    # All currently know sphinx domains.
+    #
+    # See: https://github.com/sphinx-doc/sphinx/tree/1.3/sphinx/domains
+    from sphinx.domains import c as c_domain
+    from sphinx.domains import cpp as cpp_domain
+    from sphinx.domains import javascript as javascript_domain
+    from sphinx.domains import python as python_domain
+    from sphinx.domains import std as std_domain
+
+    SPHINX_AVAILABLE = True
+except ImportError:
+    SPHINX_AVAILABLE = False
+
+# Default and/or base roles/directives to ignore.
+_base_sp_roles = ('ctype',)
+_base_sp_directives = ('autosummary', 'centered', 'currentmodule',
+                       'deprecated', 'hlist', 'include', 'index',
+                       'literalinclude', 'no-code-block', 'seealso',
+                       'toctree', 'todo', 'versionadded', 'versionchanged')
+
+
+def fetch_roles_directives():
+    """Extract all possible directives & roles that sphinx is aware of.
+
+    Returns empty lists if sphinx is not importable.
+
+    :rtype (list, list): tuple of directives and roles.
+    """
+    if not SPHINX_AVAILABLE:
+        return ([], [])
```

Comment:
Now that people are opting in to call `fetch_roles_directives`, we should probably raise an exception about how `Sphinx` wasn't found during initial import.